src/components/MultiHopTrade/hooks/useAccountIds.tsx

```diff
@@ -12,10 +18,34 @@ export const useAccountIds = (): {
 } => {
   const dispatch = useAppDispatch()
 
-  // currently, a multi-hop trade will be automagically routed though the accountId corresponding to
-  // the accountNumber for the first hop.
+  // Default sellAssetAccountId selection
+
   const sellAssetAccountId = useAppSelector(selectFirstHopSellAccountId)
-  const buyAssetAccountId = useAppSelector(selectLastHopBuyAccountId)
+  const sellAssetAccountNumberFilter = useMemo(
+    () => ({ accountId: sellAssetAccountId }),
+    [sellAssetAccountId],
+  )
+  const sellAssetAccountNumber = useAppSelector(state =>
+    selectAccountNumberByAccountId(state, sellAssetAccountNumberFilter),
+  )
+
+  // Default buyAssetAccountId selection
+
+  const accountIdsByAccountNumberAndChainId = useAppSelector(
+    selectAccountIdByAccountNumberAndChainId,
+  )
+  const inputBuyAsset = useAppSelector(selectInputBuyAsset)
+  const maybeMatchingBuyAccountId =
+    accountIdsByAccountNumberAndChainId[sellAssetAccountNumber as number]?.[inputBuyAsset?.chainId]
+
+  // We always default the buy asset account to be synchronized with the sellAssetAccountNumber
+  // - if this isn't possible, i.e there is no matching account number on the buy side, we default to the highest balance
+  // - if this was to fail for any reason, we default to the first account number as a default
+  const buyAssetAccountId = useAppSelector(state =>
+    selectLastHopBuyAccountId(state, { accountId: maybeMatchingBuyAccountId }),
+  )
```

Comment:
I'd be inclined to move all of this business logic into a new selector `selectLastHopDefaultBuyAccountId` (or similar, better name) which means the concept of maybe passing in a filter that is assumed to be `maybeMatchingBuyAccountId` is no longer required. 

i.e revert the original `selectLastHopBuyAccountId` and create a new selector encapsulating all of the above business logic.

Reply:
That was the original intent and would be much nicer, unfortunately this isn't possible because of the different arities of selectors, and the fact we can't use a selector's return value as input for the next one ☹️